sway/desktop/output.c

```diff
@@ -181,6 +181,47 @@ static void send_frame_done_iterator(struct wlr_scene_buffer *buffer,
 	}
 }
 
+static enum wlr_scale_filter_mode get_scale_filter(struct sway_output *output) {
+	switch (output->scale_filter) {
+	case SCALE_FILTER_LINEAR:
+		return WLR_SCALE_FILTER_BILINEAR;
+	case SCALE_FILTER_NEAREST:
+		return WLR_SCALE_FILTER_NEAREST;
+	default:
+		abort(); // unreachable
+	}
+}
+
+static void output_configure_scene(struct sway_output *output,
+		struct wlr_scene_node *node, float opacity) {
```

Comment:
When is `opacity` ever not 1?

Reply:
When the user configures it so? I'm not sure what you're talking about.

Reply:
> When is `opacity` ever not 1?

During recursion - it's just the initial call that is hardcoded to 1.0f.

Reply:
Oh, I missed that it was recursive, and only saw the initial call + the `static`. That makes sense, thanks.